lightning/src/ln/channelmanager.rs

```diff
@@ -3305,7 +3357,19 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 				}
 			});
 		}
-		for failure in failed_channels.drain(..) {
+		for mut failure in failed_channels.drain(..) {
+			// Channel::block_disconnected tells us to close if the funding transaction was
+			// un-confirmed due to a reorg. We cannot broadcast our latest local state via monitor
+			// update (as Channel::force_shutdown tries to make us do) as we may still be in
+			// initialization, so we track the update internally and handle it when the user next
+			// calls timer_chan_freshness_every_min, guaranteeing we're running normally.
```

Comment:
Maybe we should add a note in `timer_chan_freshness_every_min` noting that it can only be called after block sync is complete. Or add a `sync_complete` variable somewhere and assert that it's true in `timer_chan_freshness`.

Reply:
Hmm, I was trying to shove all the init comments into ChannelManagerReadArgs, I can add it here, but it also applies to all methods - I wouldnt trust calling any method prior to sync finishing, and especially not prior to moving the monitors into the chain::Watch.

Reply:
>  I wouldnt trust calling any method prior to sync finishing, and especially not prior to moving the monitors into the chain::Watch.

Ah okay. That's not crystal clear in the guide right now so I'll add that.